src/scriptengine/mod.rs

```diff
@@ -49,7 +48,17 @@ fn get_engine_type(task: &Task) -> EngineType {
                 }
             }
         },
-        None => EngineType::Unsupported,
+        None => match task.script {
+            None => EngineType::Unsupported,
+            _ => {
+                if task.script_extension.is_some() {
```

Comment:
I'm a bit worried about this case.
previously, if you didn't have a runner but you did have a script, I would attempt to execute it using sh/cmd.
now, we go to the generic script runner.
the generic script runner will panic if no shebang is defined, but if its not defined, why can't we fallback like before to sh/cmd?
So i would like a is_supported function to be exposed in the generic runner which would check if a shebang is defined, and that would be checked in this line.
if not, instead of panicking it would basically not run the script and allow the runner.rs to fallback to run it using sh/cmd.